editor/plugins/script_text_editor.cpp

```diff
@@ -273,8 +273,9 @@ void ScriptTextEditor::_set_theme_for_script() {
 	}
 }
 
-void ScriptTextEditor::_toggle_warning_pannel() {
-	warnings_panel->set_visible(!warnings_panel->is_visible());
+void ScriptTextEditor::_toggle_warning_pannel(bool p_hide) {
+	bool visible = p_hide ? false : !warnings_panel->is_visible();
```

Comment:
That's quite hacky...

Reply:
Any suggestion?
Create another function ScriptTextEditor::_hide_warning_panel(bool p_hide) and new Signal ?

Reply:
Second solution don't hide warning button and only set half opacity warning_button->set_modulate(Color(1, 1, 1, 0.5) when no warnings.

Reply:
There is probably a better way to do that, probably by adding an argument to the warning_pressed button, a boolean representing whether or not the panel should be open.

This requires storing a boolean in the code editor to know the current opening state.